cpp/src/arrow/io/compressed.cc

```diff
@@ -405,13 +412,22 @@ class CompressedInputStream::Impl {
     ARROW_ASSIGN_OR_RAISE(auto buf, AllocateResizableBuffer(nbytes, pool_));
     ARROW_ASSIGN_OR_RAISE(int64_t bytes_read, Read(nbytes, buf->mutable_data()));
     RETURN_NOT_OK(buf->Resize(bytes_read));
-    // Using std::move because the some compiler might has issue below:
+    // Using std::move because some compiler might has issue below:
```

Comment:
Can we simply try to remove the `std::move`? The linked issue seems very old.

Reply:
It's added in this issue: https://github.com/apache/arrow/issues/41024 . I don't know :-(

Reply:
Ah, thank you. It was 5 days ago, so we should keep it for now :)